src/beyond_identity/api/groups/types.rs

```diff
@@ -0,0 +1,38 @@
+use clap::Args;
+use field_types::FieldName;
+use serde::{Deserialize, Serialize};
+
+// ====================================
+// Group Structures and Types
+// ====================================
+
+#[derive(Clone, Debug, Serialize, Deserialize, FieldName)]
+pub struct Groups {
+    pub groups: Vec<GroupDetails>,
+    pub total_size: usize,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Group {
```

Comment:
I suggest renaming this to GroupEnvelope and the GroupDetails to Group.

```suggestion
pub struct GroupEnvelope {
```